src/pokemon_animation.c

```diff
@@ -861,16 +861,22 @@ u8 GetSpeciesBackAnimSet(u16 species)
 }
 
 #define tState  data[0]
-#define tPtrLO  data[1]
-#define tPtrHI  data[2]
+#define tPtrHi  data[1]
+#define tPtrLo  data[2]
 #define tAnimId data[3]
 #define tSaved0 data[4]
 #define tSaved2 data[5]
 
+#if MODERN
+#define ANIM_SPRITE(taskId)   ((struct Sprite *)((gTasks[taskId].tPtrHi << 16) | ((u16)gTasks[taskId].tPtrLo)))
+#else
+#define ANIM_SPRITE(taskId)   ((struct Sprite *)((gTasks[taskId].tPtrHi << 16) | (gTasks[taskId].tPtrLo)))
```

Comment:
Could you add a comment that explains that this is buggy under some circumstances?

Reply:
Resolved.